app/controllers/changePreferences/CheckYourAnswersController.scala

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.changePreferences
+
+import cats.data.EitherT
+import com.google.inject.Inject
+import config.Constants.submissionDetailsKey
+import controllers.actions.{DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import controllers.routes
+import models._
+import play.api.Logging
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmailVerificationService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import utils.{CheckYourAnswersSummaryListHelper, PageCheckHelper}
+import views.html.changePreferences.CheckYourAnswersView
+
+import java.time.Instant
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckYourAnswersController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  pageCheckHelper: PageCheckHelper,
+  summaryListHelper: CheckYourAnswersSummaryListHelper,
+  emailVerificationService: EmailVerificationService,
+  val controllerComponents: MessagesControllerComponents,
+  view: CheckYourAnswersView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport
+    with Logging {
+
+  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    pageCheckHelper.checkDetailsForCheckYourAnswers(request.userAnswers) match {
+      case Right(mustCheckVerificationStatus) =>
+        if (mustCheckVerificationStatus) {
+          val enteredEmail = request.userAnswers.emailAddress.getOrElse(
+            throw new RuntimeException("Entered email address missing from user answers")
```

Comment:
I would generally use an Illegal Argument / Sate Exception here to be slightly more specific but I am not so precious.